public/JSONtoHTML.coffee

```diff
@@ -0,0 +1,211 @@
+# Variables
+xmlHttp = new XMLHttpRequest
+jsonFile = 'stdLib_base_test.json'
+
+getMainData = (data) ->
+  output = ''
+  modules = data.modules
+
+  i = 0
+  while i < modules.length
+    module = modules[i]
+    output += '<div class="moduleDiv" id="'+ module.name + '">'
+    output += '<div style="width:100%" onMouseOver="document.getElementById(\''+module.name+'_Hover\').style.display = \'inline-block\'" onMouseOut="document.getElementById(\''+module.name+'_Hover\').style.display = \'none\'"><p class="module">' + module.name + '</p>'
+    if module.tag != null && module.tag != undefined 
+      tagType = module.tag.split(":")
+      output += '<div class="tag tag_' + tagType[0].replace(/\s/g,'') + '"><p class="<!--fa fa-tags-->">' + tagType[0].replace(/\s/g,'') + ' in ' + tagType[1].replace(/\s/g,'')+ '</p></div>'
+    output += '<div id="'+module.name+'_Hover" class="hoverableAnchors"><a href="#' + module.name + '" class="fa fa-anchor" title="Get anchor link"></a>&nbsp;&nbsp;&nbsp;&nbsp;<a href="" class="fa fa-file-code" title="Jump to source"></a></div>'
+    output += '</div>'
+
+    if module.documentation != null && module.documentation != undefined
+      mdtohtml = marked(module.documentation)
+      output += '<p class="module_overview">Module overview</p>'
+      output += '<div class="moduleDocumentation">' + mdtohtml + '</div>'
+
+    output += '<div class="module_classes">'
+    j = 0
+    while j < module.classes.length
+      Mclass = module.classes[j]
+      output += '<div class="module_classes_class" id="' + module.name + '_' + Mclass.name + '"><div style="width:100%" onMouseOver="document.getElementById(\'' + module.name + '_' + Mclass.name + '_Hover\').style.display = \'inline-block\'" onMouseOut="document.getElementById(\'' + module.name + '_' + Mclass.name + '_Hover\').style.display = \'none\'"><p class="className">' + Mclass.name + '</p>'
+      if Mclass.tag != null && Mclass.tag != undefined 
+        tagType = Mclass.tag.split(":")
+        output += '<div class="tag tag_' + tagType[0].replace(/\s/g,'') + '"><p class="<!--fa fa-tags-->">' + tagType[0].replace(/\s/g,'') + ' in ' + tagType[1].replace(/\s/g,'')+ '</p></div>'
+      output += '<div id="' + module.name + '_' + Mclass.name + '_Hover" class="hoverableAnchors"><a href="#' + module.name + '_' + Mclass.name + '" class="fa fa-anchor"></a>&nbsp;&nbsp;&nbsp;&nbsp;<a href="" class="fa fa-file-code"></a></div></div>'
+     
+      if Mclass.documentation != null
+        mdtohtml = marked(Mclass.documentation)
+        output += '<div class="classDocumentation">' + mdtohtml + '</div>'
+
+      if Mclass.longDocumentation != null && Mclass.longDocumentation != undefined
+        mdtohtml = marked(Mclass.longDocumentation)
+        output += '<div class="classLongDocumentation">' + mdtohtml + '</div>'  
+
+      output += '<div class="module_classes_class_methods">'
+      k = 0
+      printedInfo = false
+      while k < Mclass.methods.length
```

Comment:
Why don't you use other, more coffee-style loops instead of while and custom incrementation?